descqa/AngularCorrelation.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import print_function, division, unicode_literals, absolute_import
+import os
+import math
+try:
+    from itertools import zip_longest
+except ImportError:
+    from itertools import izip_longest as zip_longest
+
+import numpy as np
+from GCR import GCRQuery
+
+from .base import BaseValidationTest, TestResult
+from .plotting import plt
+import treecorr
+
+__all__ = ['AngularCorrelation']
+
+
+
+
+class AngularCorrelation(BaseValidationTest):
+    """
+    Validation test to show 2pt correlation function of ProtoDC2 with SDSS
+    """
+    def __init__(self, band='r', survey='sdss', observation='', 
+                 RandomFactor=10, **kwargs):
+
+        #catalog quantities
+        self.mag_field = 'mag_{}_{}'.format(band, survey)
+        self.band = band
+        self.redshift_max = 0.3
+
+        #validation data
+        self.validation_data = {}
+        self.observation = observation
+
+        #Random number times shape of catalog
+        self.RandomFactor = RandomFactor
+
+        #setup summary plot
+        self.first_pass = True
+
+        self._other_kwargs = kwargs
+
+        self.possible_observations = {
+            'Zehavi2011_rAbsMagSDSS': {
+                'filename_template': '2pt/Zehavi2011_SDSS_r_{}.dat',
+                'usecols': (0, 1),
+                'colnames': ('rp', 'wp'), #This is not the names of the columns in the data files but the name assigned to the columns 0 and 1
+                'skiprows': 1,
+                'label': 'Zehavi et al 2011',
+            },
+            'Wang2013_rAppMagSDSS': {
+                'filename_template': '2pt/Wang2013_SDSS_fig15_r_{}.dat',
+                'usecols': (0, 1),
+                'colnames': ('theta', 'w'),
+                'skiprows': 1,
+                'label': 'Wang et al 2013',
+            },
+        }
+
+
+    @staticmethod
+    def get_catalog_data(gc, quantities, filters=None):
+        '''
+        Get catalog data
+        '''
+        data = {}
+        if not gc.has_quantities(quantities):
+            return TestResult(skipped=True, summary='Missing requested quantities')
```

Comment:
change to `return None`
Only `get_validation_data` would return `TestResult`

Reply:
Should line 70 be changed to `return None`?

Reply:
Yes. And later you can check `if catalog_data is None` to decide whether to skip the test.